plugins/scale/scale.cpp

```diff
@@ -1082,15 +1112,27 @@ class wayfire_scale : public wf::per_output_plugin_instance_t,
     {
         if (scale_data.count(get_top_parent(view)) != 0)
         {
-            remove_view(view);
-            if (scale_data.empty())
+            if (include_minimized && view->minimized && should_scale_view(view))
```

Comment:
I think we should be really careful here, view_disappeared is not emitted only when the view is minimized. This path here could be triggered if for example a minimized view is closed (for example by killing the client).

I tested this locally and it doesn't actually cause any issues because `should_scale_view()` returns false for unmapped views - but I thought it might be worth adding a comment for future reference :)

Reply:
Yes, I've run into this issue while testing -- I was experimenting if there is a more efficient way to distinguish between these two cases than `should_scale_view()`, but decided that this is likely the most robust one.